File: UIA_Warm_Up/CPP/chatroom.cpp
#include <iostream>
#include <string>

using namespace std;


int main(){
	string word = "hello";
	string found = "";
	string got;
	cin>>got;
	int j = 0;

	for (int i = 0; i < word.size(); ++i)
	{
		char letter = word[i];
		for (j; j < got.size(); ++j)
		{
			if (got[j] == letter){
				found+=got[j];
				j+=1;
				break;
			}
		}
	}

	if (found == word)
		cout<<"YES"<<endl;
	else
		cout<<"NO"<<endl;
	return 0;
}

File: practice/Chat Room.py
word = "hello"
got = input()

res = ""
j=0

for i in range(5):
	for j in range(j, len(got)):
		if got[j] == word[i]:
			res+= got[j]
			j+=1
			break
if res == word:
	print("YES")
else:
	print("NO")
